src/kinetics/GasKinetics.cpp

```diff
@@ -266,6 +266,15 @@ bool GasKinetics::addReaction(shared_ptr<Reaction> r)
         return true;
     }
 
+    if (!addSpecificReaction(r)) {
+        throw CanteraError("GasKinetics::addReaction",
+            "Unknown reaction type specified: '{}'", r->type());
+    }
+    return true;
+}
+
+bool GasKinetics::addSpecificReaction(shared_ptr<Reaction> r)
+{
```

Comment:
All reaction types should use a conventional factory loader (i.e. no specialized handlers).